pysat/utils/registry.py

```diff
@@ -84,48 +111,154 @@ def register(module_name):
     pysat instrument files could result in unexpected consequences.
 
     Examples
-    ---------
-    from pysat import Instrument, user_modules
-    from pysat.utils import registry
+    --------
+    ::
+        from pysat import Instrument, user_modules
+        from pysat.utils import registry
 
-    registry.register('my.package.name.myInstrument')
-    assert 'my.package.name.myInstrument' in user_modules
+        registry.register('my.package.name.myInstrument')
+        assert 'my.package.name.myInstrument' in user_modules
 
+        testInst = Instrument(platform, name)
 
+    """
 
-    testInst = Instrument()
+    if isinstance(module_names, str):
+        module_names = [module_names]
 
+    for module_name in module_names:
+        # first, ensure module string directs to something importable
+        try:
+            inst_module = importlib.import_module(module_name)
+        except Exception as error:
+            # log error and then preserve trace and propagate error
+            logger.error(error)
+            raise
+
+        # second, check that module is itself pysat compatible
+
```

Comment:
This is where the minimum feature set for a pysat instrument needs to go.

Reply:
pinging @jklenzing since you are closest to the updated test structure.

Reply:
A couple options are in the code

Reply:
If there are no other opinions I prefer the second option.

 Also thinking we can write a method in utils, something like verify_pysat_standard (compatibility?). Input a instrument module, get a report on how it fares against pysat standards, and test against methods. Basically a simple way to test things in development.

Reply:
> A couple options are in the code

It may be more useful for discussion to put them in the conversation here as examples or put them on a slide/wiki page as examples to be discussed at the next meeting.

Reply:
Good point.

prep work in containing method.
```
inst_module = importlib.import_module(module_name)      
validate = itc.InstTestClass()
```

First option, using the test class as is.
```
        class Foo(object):
            pass
        # parse string to get package part and instrument module part
        parse = module_name.split('.')

        validate.package = Foo()
        validate.package.__name__ = '.'.join(parse[:-1])
        validate.test_modules_standard(parse[-1])
        validate.test_standard_function_presence(parse[-1])
```
No modifications to the testing code is required. Requires a dummy class and relation of prep code to use by validate isn't clear unless familiar with validate.

Using the test class with a slight mod for outside use. Doesn't impact use of current test code.
```
        validate.test_modules_standard('', inst_module)
        validate.test_standard_function_presence('', inst_module)
```
I like the direct input of the module under test, though it comes with the downside of having a '' leading input and changes to input of test methods.

Reply:
I went ahead and cleaned things up in favor of the first option and using the code as it exists. Not worth adding in an optional code path for something there is already have a solution for. I spent a bit of time on the comments instead.